packages/react-slider/src/components/Slider/useSliderStyles.ts

```diff
@@ -4,297 +4,135 @@ import type { SliderState } from './Slider.types';
 
 export const sliderClassName = 'fui-Slider';
 
-export const thumbClassName = `${sliderClassName}-thumb`;
-export const trackClassName = `${sliderClassName}-track`;
-export const markContainerClassName = `${sliderClassName}-markItemContainer`;
-export const firstMarkClassName = `${sliderClassName}-firstMark`;
-export const lastMarkClassName = `${sliderClassName}-lastMark`;
-export const markClassName = `${sliderClassName}-mark`;
-export const markLabelClassName = `${sliderClassName}-label`;
+const thumbSizeVar = `--fui-slider-thumb-size`;
+const railSizeVar = `--fui-slider-rail-size`;
+const railColorVar = `--fui-slider-rail-color`;
+const progressColorVar = `--fui-slider-progress-color`;
+const thumbColorVar = `--fui-slider-thumb-color`;
+
+export const railDirectionVar = `--fui-slider-rail-direction`;
+export const railOffsetVar = `--fui-slider-rail-offset`;
+export const railProgressVar = `--fui-slider-rail-progress`;
+export const railStepsPercentVar = `--fui-slider-rail-steps-percent`;
+export const thumbPositionVar = `--fui-slider-thumb-position`;
+
+const get = (value: string) => `var(${value})`;
 
 /**
  * Styles for the root slot
  */
 export const useRootStyles = makeStyles({
-  root: theme => ({
+  root: {
     position: 'relative',
-    display: 'inline-flex',
+    display: 'inline-grid',
+    gridTemplateAreas: '"slider"',
     userSelect: 'none',
     touchAction: 'none',
-    verticalAlign: 'bottom',
-  }),
+  },
 
   small: {
-    '--slider-thumb-size': '10px',
-    '--slider-rail-size': '2px',
-    '--slider-mark-size': '2px',
+    [thumbSizeVar]: '16px',
+    [railSizeVar]: '2px',
   },
 
   medium: {
-    '--slider-thumb-size': '20px',
-    '--slider-rail-size': '4px',
-    '--slider-mark-size': '4px',
+    [thumbSizeVar]: '20px',
+    [railSizeVar]: '4px',
   },
 
-  horizontal: theme => ({
+  horizontal: {
     minWidth: '120px',
-    minHeight: 'var(--slider-thumb-size)',
-    flexDirection: 'column',
-  }),
+    height: get(thumbSizeVar),
+    alignItems: 'center',
+  },
 
-  vertical: theme => ({
-    transform: 'scaleY(-1)',
-    minWidth: 'var(--slider-thumb-size)',
+  vertical: {
+    width: get(thumbSizeVar),
     minHeight: '120px',
-    flexDirection: 'row',
-  }),
+    justifyItems: 'center',
+    gridTemplateColumns: get(thumbSizeVar),
+  },
 
   enabled: theme => ({
-    cursor: 'grab',
+    [railColorVar]: theme.colorNeutralStrokeAccessible,
```

Comment:
This type of change is very likely to clash with #20918. If that goes in first let me know if you want me to push an update to your branch and I'll be happy to do it.

Reply:
yeah, this will cause some re-work to be done here. if the PR is stable, I can pull it in and start making the updates now